Add contract verification guide to Developing Contracts

ElPaisano opened the pull request:
Addresses https://github.com/filecoin-project/filecoin-docs/pull/1952#issuecomment-1551015334 by adding a basic guide on contract verification to Developing Contracts. Based off of https://gist.github.com/trruckerfling/54918b8331f6fa10ae27cedd6afd9680

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+
+1. In the **File Explorer** sidebar tab, under **contracts**, right click on your contract.
+
+1. From the menu, select **generate UM** to flatten `<contract-name>.sol` file and ensure all components and tasks are included. 
```
adding screenshot:
<img width="779" alt="Screen Shot 2023-04-21 at 4 56 01 PM" src="https://user-images.githubusercontent.com/113331491/233750337-b15e51b3-aad5-4a50-bf62-6fc6625799d2.png">

ElPaisano replied:
> edit to "generate UML"

Done ✅

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+1. In the form, enter the following information:
+   - The address of your deployed contract.
+   - The compiler version used in 
+   - The license type used in
```
Adding screenshot:

<img width="776" alt="Screen Shot 2023-04-21 at 5 05 56 PM" src="https://user-images.githubusercontent.com/113331491/233750902-863385da-8325-43c0-b516-6597adaeb54b.png">

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+1. Click **Continue** to upload your flattened .sol file. 
+
+1. Click **Verify and Publish**.
+
```
Adding screenshot:

<img width="776" alt="Screen Shot 2023-04-21 at 5 07 39 PM" src="https://user-images.githubusercontent.com/113331491/233750996-00bfeff1-e89e-4578-ae98-f30659bc6e92.png">

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+1. Click **Verify and Publish**.
+
+   Success! Your contract is now verified.
+
```
Adding screenshot: 

<img width="776" alt="Screen Shot 2023-04-21 at 5 08 47 PM" src="https://user-images.githubusercontent.com/113331491/233751035-d3ae7a4e-d03e-419b-8958-e4109df9b9eb.png">

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+   Success! Your contract is now verified.
+
+1. In Filfox's search bar, enter your contract address.
+
```
Adding screenshot:

<img width="1387" alt="Screen Shot 2023-05-24 at 3 17 04 PM" src="https://user-images.githubusercontent.com/113331491/240517763-6681b22f-603b-4b56-98e1-5489bd33327e.png">

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+1. In Filfox's search bar, enter your contract address.
+
+1. Select the **Contract** tab to you can view your contract's source code and ABI.
+
```
<img width="1387" alt="Screen Shot 2023-05-24 at 3 21 02 PM" src="https://user-images.githubusercontent.com/113331491/240518584-e162c17d-3656-4577-96e0-11c72d65415f.png">

trruckerfling commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+
+1. Select the **Contract** tab to you can view your contract's source code and ABI.
+
+> You may also view other verified contracts' source code and ABI by heading back to Filfox contract verification page and clicking on 'Verified Contract List'. You can click into contract addresses and follow the steps above, in this section.
```
![Screen Shot 2023-05-24 at 3 23 46 PM](https://user-images.githubusercontent.com/113331491/240519458-a4838f5f-ab50-4420-8203-22fedd090161.png)
<img width="1387" alt="Screen Shot 2023-05-24 at 3 25 20 PM" src="https://user-images.githubusercontent.com/113331491/240519733-43217202-c75f-480a-8eee-79c964fd3c7d.png">

ElPaisano commented:
> Added screenshots to include and light edits for accuracy. Please help review and merge, thank you!

@trruckerfling sorry for the confusion here, this PR was half-baked when you reviewed.

Finished up the PR. Rewrote overall flow and addressed some of your requested changes:

- "UM" -> "UML"
- Fixed the frontmatter `lead`
- Renamed sections

I tried to write the instructions in such a way that screenshots are not needed. Generally, I don't like screenshots in the docs because it generally means more maintenance since screenshots and UIs get out of date fast. So, I'd prefer not to include screenshots unless we feel it's absolutely needed.

ElPaisano commented:
@johnnymatthews a few things here for you:

- what do you think RE https://github.com/filecoin-project/filecoin-docs/pull/1993#issuecomment-1569051392?
- where should this page live in the docs hierarchy? It's currently in Smart contracts -> developing contracts

Also, note that for merge purposes, this PR spun out of [https://github.com/filecoin-project/filecoin-docs/pull/1952#issuecomment-1551015334](https://github.com/filecoin-project/filecoin-docs/pull/1952#issuecomment-1551015334)

ElPaisano commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+
+First
+
+1. Open Remix.
```
Hey @trruckerfling this is a Markdown convention that the docs team (me and @johnnymatthews lol) try and keep across the docs. It renders as 1, 2, 3, ... in the UI

ElPaisano commented on content/en/smart-contracts/developing-contracts/verify-contracts/index.md:
```diff
@@ -0,0 +1,72 @@
+---
+
+## Verify your contract
+If you are deploying your FEVM contract to mainnet, we highly recommend verifying your contract, to upkeep security and transparency on the network.
```
Addressed

trruckerfling reviewed (approved):
good to go!

johnnymatthews reviewed (approved):
Looks good!
